msal-dotnet-articles/advanced/monitoring.md

```diff
@@ -5,44 +5,151 @@ description: "How to use the metrics provided by MSAL.NET to monitor your applic
+1. Added the [OpenTelemetry](https://www.nuget.org/packages/OpenTelemetry#readme-body-tab) package dependency to your project.
+1. Added an exporter dependency, that will allow you to export logs. One such example is the [Console exporter for OpenTelemetry.NET](https://www.nuget.org/packages/OpenTelemetry.Exporter.Console/1.7.0-alpha.1).
 
-> [!NOTE]
-> Regionalization is available only to internal Microsoft applications for now.
+>[!NOTE]
+>While the console exporter is a good start for local debugging and diagnostics, it's not the best choice for production-deployed applications. We recommend checking out the [official exporter documentation](https://opentelemetry.io/docs/instrumentation/net/exporters/) to learn more about available options. If you are hosting applications on Azure, you may consider ingesting OpenTelemetry data in [Azure Data Explorer](/azure/data-explorer/open-telemetry-connector?tabs=command-line) or [Azure Monitor](/azure/azure-monitor/app/opentelemetry-enable?tabs=aspnetcore).
+
+In your application initialization code, prior to bootstrapping the MSAL authentication client (e.g., PublicClientApplication or ConfidentialClientApplication), declare a new MeterProvider instance, like this:
+
+```csharp
+using var meterProvider = Sdk.CreateMeterProviderBuilder()
+    .AddMeter("MicrosoftIdentityClient_Common_Meter")
+    .AddConsoleExporter()
+    .Build();
+```
+
+This will initialize the meter provider and use the built-in MSAL.NET meter (`MicrosoftIdentityClient_Common_Meter`) that captures a series of counters and histograms. When a console exporter is used, you should see the output being piped directly in the terminal:
+
+![Example of OpenTelemetry outputting metrics to the terminal](../media/msal-net-logging/opentelemetry-logging.gif)
+
+The following section outlines the supported counters and histograms for the default meter.
+
+### Counters
+
+#### `msalsuccess_counter`
+
+Counter to capture aggregation of successful requests in MSAL.
+
+##### Metadata
+
+| Field              | Description |
+|:-------------------|:------------|
+| MsalVersion        | Version of MSAL used. |
+| Platform           | .NET SKU used.        |
+| ApiId              | ID for the API used for token acquisition. |
+| TokenSource        | Source of token (e.g., identity provider or cache). |
+| CacheRefreshReason | Reason for cache refresh. |
+| CacheLevel         | L1, L2, or Unknown when the custom cache is used but level is not recorded. |
+
+#### `msalfailure_counter`
+
+Counter to capture aggregation of failed requests in MSAL.
+
+##### Metadata
+
+| Field              | Description |
+|:-------------------|:------------|
+| MsalVersion        | Version of MSAL used. |
+| Platform           | .NET SKU used.        |
+| ErrorCode          | Microsoft Entra ID error code in case of MsalServiceException, MsalErrorCode in case of MsalClientException or name of the exception in case it is not an MsalException. |
```

Comment:
```suggestion
| `MsalVersion`        | Version of MSAL used. |
| `Platform`           | .NET SKU used.        |
| `ErrorCode`          | Microsoft Entra ID error code in case of `MsalServiceException`, `MsalErrorCode` in case of `MsalClientException` or name of the exception in case it isn't an `MsalException`. |
```